solr/core/src/java/org/apache/solr/search/ReRankCollector.java

```diff
@@ -129,21 +129,25 @@ public TopDocs topDocs(int start, int howMany) {
 
       ScoreDoc[] mainScoreDocs = mainDocs.scoreDocs;
       ScoreDoc[] mainScoreDocsClone =
-          (reRankScaler != null && reRankScaler.scaleScores())
-              ? deepCloneAndZeroOut(mainScoreDocs)
-              : null;
+          deepClone(mainScoreDocs, reRankScaler != null && reRankScaler.scaleScores());
```

Comment:
boolean args can be unclear at the call site.  To mitigate that, please declare a local var zeroOutScores initialized to this and pass in.

Reply:
I agree, change is coming in the next commit